addons/addon-base-raas/packages/base-raas-services/lib/environment/service-catalog/environment-sc-service.js

```diff
@@ -271,6 +275,89 @@ class EnvironmentScService extends Service {
     return result;
   }
 
+  async changeWorkspaceRunState(requestContext, { id, operation }) {
+    const existingEnvironment = await this.mustFind(requestContext, { id });
+
+    // Make sure the user has permissions to change the environment run state
+    await this.assertAuthorized(
+      requestContext,
+      { action: 'update-sc', conditions: [this._allowAuthorized] },
+      existingEnvironment,
+    );
+
+    const { status, outputs, projectId } = existingEnvironment;
+
+    // expected environment run state based on operation
+    const expectedStatus = operation === 'start' ? 'STOPPED' : 'COMPLETED';
+    if (status !== expectedStatus) {
+      throw this.boom.badRequest(
+        `unable to ${operation} environment with id "${id}" - current status "${status}"`,
+        true,
+      );
+    }
+    let instanceType;
+    let instanceIdentifier;
+    const outputsObject = cfnOutputsToObject(outputs);
```

Comment:
outputsObject is a dictionary. Let's name it something like 'outputsDict'

Reply:
Emmmm there is no 'dictionary' type in JS, a JS object is essentially key value pairs that can act as a dictionary. I'll update the function name to be something like cfnOutputsArrayToObject()?

Reply:
Since this would be clear to people working with JS, I wouldn't mind this being left as-is. Thanks for clarifying